### PR TITLE
Make Debsources Responsive

### PR DESCRIPTION
**Pull request by devoxel**
Note that this is not ready to actually be pulled yet, but opening the PR allows people to easily keep an eye on changes without having to dig through stuff or host it on their local machine.

http://sourcesdev.debian.net should usually have an up to date hosted version of this, although recent commits may take time to pull.

### CODE REVIEW

File: debsources/app/copyright/templates/copyright/index.html [hunk 1]
```diff
@@ -4,7 +4,8 @@
   https://anonscm.debian.org/gitweb/?p=qa/debsources.git;a=blob;f=AUTHORS;hb=HEAD
   License: GNU Affero General Public License, version 3 or above.
 #}
-{# copied from templates/index.html #}
+{# XXX copied from templates/index.html #}
+{# this should be a macro: repetitive code which makes change awkward #}
```
Comment (oorestisime) [inline debsources/app/copyright/templates/copyright/index.html:8]: Please remove unnecessary comments from the code :)

File: debsources/app/copyright/templates/copyright/index.html [hunk 2]
```diff
@@ -15,31 +16,43 @@
...
+      </sup>
+    </h3>
+
+    <div class="col-xs-2"></div>
+    <div class="col-xs-8">
+      <p>Browse through the licenses of the
+      <a href="https://www.debian.org">Debian</a> operating system.
+      <a href="{{ url_for('doc.doc_overview') }}">Read more…</a></p>
+    </div>
+    <div class="col-xs-2"></div>
```
Comment (oorestisime) [inline debsources/app/copyright/templates/copyright/index.html:37]: if what you are trying to achieve is centering the previous thing then you should have a look at offset/pull/push classes of columns that enable this with less and clearer code. Unless i am missing something, please mention so :)

File: debsources/app/copyright/templates/copyright/license_render.inc.html [hunk 3]
```diff
@@ -1,69 +1,91 @@
...
+      <div id="desc-{{ loop.index0 }}">
+        <pre class='r_synopsis'>{{ license['text'] }} </pre>
+        {%- if license['comment'] is not none %}
+          <p class='r_l_comment'><span class='r_decorate'>Comment: </span>{{ macro.match_d_license(license['comment']) }}</p>
+        {%- endif %}
+      </div>
+    {%- endfor %}
+  </div>
+
+</div>
```
Comment (oorestisime) [inline debsources/app/copyright/templates/copyright/license_render.inc.html:91]: Looking at this on sourcesdev feels like difficult to use.. the table has a horizontal scroll (default bootstrap behavior) but its pretty much unusable. I'd look into hiding a column on small screens. for example the comment column.
Not a blocker for me just dropping some ideas

File: debsources/app/templates/news_archive.html [hunk 4]
```diff
@@ -14,24 +14,41 @@
...
 </div>
+
 {% endblock %}
 
-{# show the background debian image #}
+{# XXX: This is the old way bg_wrapping is handled. The new modified way uses
+        super() to change the image.
+        If this is rejected this block of code will be needed.
+
+show the background debian image
```
Comment (oorestisime) [inline debsources/app/templates/news_archive.html:35]: i don't think we need comments like this in the templates. i you want you can add these issues on trello so we can see them as actionable items etc.

File: debsources/app/templates/footer.inc.html [hunk 5]
```diff
@@ -1,34 +1,44 @@
...
+  <div class="col-md-12">
+    <p style="margin: 0 0 0 0; line-height: 1em;">
+      Browse by prefix: &ensp;
+      {{ macros.render_packages_prefixes(packages_prefixes, request) }}
+      &ensp; | &ensp;
+      Browse <a href="{{ url_for(endpoint, page=1) }}">by page</a>
+    </p>
+  </div>
+  <hr />
+  <!-- XXX fix: visability of this-->
```
Comment (oorestisime) [inline debsources/app/templates/footer.inc.html:19]: leftover....

File: debsources/app/templates/_macros.html [hunk 6]
```diff
@@ -37,17 +37,20 @@
...
         method="post" style="display: {{ display }};">
       {{ form.query(**kwargs) }}
-    {% for error in form.errors.query %}
-      {% if display == "block" %}<br />{% endif %}
-      <strong>{{ error }}</strong>
-    {% endfor %}
     {% if display == "block" %}<br />{% endif %}
     <input type="submit" value="Search package" />
-  </form>
+  </form>
```
Comment (oorestisime) [inline debsources/app/templates/_macros.html:53]: spaces :))

**Comment by oorestisime** [comment]
Heya, 

i reviewed a bit your PR. Generally good work nice.
Besides the comments above here are some general notes:
- some XXX fix this ... leftovers in the templates. please move them on to trello or delete them from there.. we should keep the templates clean
- i guess you use a tool that autoformats or so? there are many changes that involve just adding a blank line somewhere or making one liner html into many lines. maybe its better to avoid this
- i would leave debian.css unmodified. If you need to overwrite styles then you should probably do it in base.css like you did for boostrap. that way if a new version of debian.css comes out it will be trivial to update

Haven't taken the time yet to test this locally but i pulled on sourcesdev. The tests locally are ok for you?
In any case keep up the good work :)

**Comment by devoxel** [comment]
Thanks for review. I already wrote a reply to this but lost it somehow... Anyways:

> some XXX fix this ... leftovers in the templates. please move them on to trello or delete them from there.. we should keep the templates clean

I agree. Will do.

> i guess you use a tool that autoformats or so? there are many changes that involve just adding a blank line somewhere or making one liner html into many lines. maybe its better to avoid this

I don't have an autoformatting tool, I changed some things I felt were indented badly. However, if you disagree with specific instances I will certainly be open to change.

> i would leave debian.css unmodified. If you need to overwrite styles then you should probably do it in base.css like you did for boostrap. that way if a new version of debian.css comes out it will be trivial to update

If I recall correctly there was already some debsources specific stuff in there. However, you're right really, it's the better standard thing to do. I'll change over to that.

The only tests failing are failing on travis only and in master, as mentioned on the IRC. Nothing else is failing for me.

File: debsources/app/copyright/templates/copyright/index.html (hunk 2 above)
Comment (devoxel) [inline debsources/app/copyright/templates/copyright/index.html:37]: Fixed this, will push up eventually!

File: debsources/app/copyright/templates/copyright/index.html (hunk 1 above)
Comment (devoxel) [inline debsources/app/copyright/templates/copyright/index.html:8]: Fixed this, will push up soon to the related branches. I'll review diffs and make sure there's not much more

**Pull request closed without merging**